mono/metadata/gc.c

```diff
@@ -681,6 +682,31 @@ ves_icall_System_GCHandle_GetTarget (guint32 handle)
 	return mono_gchandle_get_target (handle);
 }
 
+static gboolean
+is_object_pinnable (MonoObject *obj)
```

Comment:
`Why is it giving special treatment to char and boolean?`
I believe because they are pinnable but not blittable since they are marshaled to a native type of different size.